zeth-contracts/contracts/BaseMerkleTree.sol

```diff
@@ -70,4 +70,13 @@ contract BaseMerkleTree {
     // Returns the array of leaves of the merkle tree
     return tmpLeaves;
   }
+
+  // Retrieve a single tree entry.  We expect that clients will generally
+  // decrypt a very small fraction of new entris that appear in the tree, which
+  // represents an even smaller fraction of the total tree, so in general it is
+  // very inefficient for clients to download the entire tree.
+  function getLeaf(uint leaf_address) public view returns (bytes32) {
```

Comment:
Where do you use this @dtebbs ?

That quite a dangerous operation to carry out on the tree as it can leak deadly metadata about the commitments you are accessing in the tree!

Reply:
Manipulating the tree in its entirety is less efficient, not to say very inefficient, but it prevents several types of leakages since you never really know which entries "interest" the user

Reply:
This has now been removed (as discussed below)